### PR TITLE
Migrate to prow

### PR DESCRIPTION
**Pull request by dashpole**
Implements changes in cAdvisor to use:
https://github.com/kubernetes/test-infra/pull/6302

cc @krzyzacy

### CODE REVIEW

File: build/jenkins_e2e.sh [hunk 1]
```diff
@@ -61,7 +61,15 @@ golden_nodes=(
...
+if [[ -a "${CADVISOR_SERVICE_ACCOUNT:-}" ]] ; then
+  gcloud auth activate-service-account --key-file="${CADVISOR_SERVICE_ACCOUNT}"
+fi
+
 ./runner --logtostderr --test-retry-count=$max_retries \
   --test-retry-whitelist=integration/runner/retrywhitelist.txt \
-  --ssh-options "-i /var/lib/jenkins/gce_keys/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no" \
+  --ssh-options "--project=cadvisor-e2e --strict-host-key-checking=false" \
   ${golden_nodes[*]}
+
```
Comment (krzyzacy) [inline build/jenkins_e2e.sh]: nit: add a comment say we need to switch back using this service account for uploading logs etc

**Comment by BenTheElder** [comment]
/retest
🤷‍♂️

**Comment by BenTheElder** [comment]
@krzyzacy we probably want to fix tot? 🤷‍♂️

**Comment by krzyzacy** [comment]
```
I0117 22:29:28.446] Call:  git fetch --quiet --tags https://github.com/kubernetes/cadvisor master +refs/pull/1868/head:refs/pr/1868
W0117 22:29:28.682] fatal: could not read Username for 'https://github.com': No such device or address
```
whoops it's not in kubernetes
/shrug

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/pull/6320

**Comment by BenTheElder** [comment]
> whoops it's not in kubernetes

https://github.com/kubernetes/test-infra/pull/6320

**Comment by BenTheElder** [comment]
> W0117 22:43:46.722] build/assets.sh: line 31: go: command not found

We should move this to the kubekins-master image.. opening another PR

**Comment by BenTheElder** [comment]
@dashpole 
```bash
 # Build & test with go 1.8
 docker run --rm \
        -w "/go/src/github.com/google/cadvisor" \
        -v "${GOPATH}/src/github.com/google/cadvisor:/go/src/github.com/google/cadvisor" \
        golang:1.8 make all test-runner
```
can we replace this with something like:
```bash
export GOPATH=/go
CADVISOR_GOPATH=$GOPATH/src/github.com/google/cadvisor
mkdir -p $CADVISOR_GOPATH
ln -s $CADVISOR_GOPATH $PWD
make all test-runner
```
We will have an image with go installed running this job, so mostly we just need the $GOPATH to be right. (xref: https://github.com/kubernetes/test-infra/pull/6322)

**Comment by dashpole** [comment]
@BenTheElder what version of go will the image have?

**Comment by BenTheElder** [comment]
@dashpole 1.9.2 currently but we maintain a variant of the image for each k8s release branch so we could get EG 1.8.3 by switching to the k8s 1.8 or 1.7 image.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/blob/f6546fbe531316615649aca5a43b6c310772692d/images/kubekins-e2e/Makefile#L18-L47

**Comment by dashpole** [comment]
sgtm

**Comment by krzyzacy** [comment]
https://prow.k8s.io/log?job=pull-cadvisor-e2e&id=5 ^^

**Comment by BenTheElder** [comment]
Hmmm... `kubekins-e2e` does:
```
ENV GOPATH /go
ENV PATH /usr/local/go/bin:$PATH
ENV PATH $GOPATH/bin:$PATH

# setup k8s repo symlink
RUN mkdir -p /go/src/k8s.io/kubernetes \
    && ln -s /go/src/k8s.io/kubernetes /workspace/kubernetes
```

So we probably shouldn't set $GOPATH but i'm not sure why `go list ./...` is giving `_/workspace/...` results.

File: build/jenkins_e2e.sh [hunk 2]
```diff
@@ -61,7 +61,15 @@ golden_nodes=(
...
 max_retries=8
 
+if [[ -a "${CADVISOR_SERVICE_ACCOUNT:-}" ]] ; then
+  gcloud auth activate-service-account --key-file="${CADVISOR_SERVICE_ACCOUNT}"
+fi
+
 ./runner --logtostderr --test-retry-count=$max_retries \
   --test-retry-whitelist=integration/runner/retrywhitelist.txt \
-  --ssh-options "-i /var/lib/jenkins/gce_keys/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no" \
+  --ssh-options "--project=cadvisor-e2e --strict-host-key-checking=false" \
```
Comment (krzyzacy) [inline build/jenkins_e2e.sh]: `--strict-host-key-checking=no`

**Comment by BenTheElder** [comment]
https://cloud.google.com/sdk/gcloud/reference/compute/ssh
`--strict-host-key-checking=` can be one of `yes`, `no`, `ask`. We probably don't even need to set this flag?
>  By default, StrictHostKeyChecking is set to 'no' the first time you connect to an instance, and will be set to 'yes' for all subsequent connections.

**Comment by dashpole** [comment]
We had it set previously... maybe I should try it without first though

**Comment by krzyzacy** [comment]
probably that's a change from gcloud side

File: build/jenkins_e2e.sh [hunk 3]
```diff
@@ -61,7 +61,15 @@ golden_nodes=(
...
 max_retries=8
 
+if [[ -a "${CADVISOR_SERVICE_ACCOUNT:-}" ]] ; then
+  gcloud auth activate-service-account --key-file="${CADVISOR_SERVICE_ACCOUNT}"
+fi
+
 ./runner --logtostderr --test-retry-count=$max_retries \
   --test-retry-whitelist=integration/runner/retrywhitelist.txt \
-  --ssh-options "-i /var/lib/jenkins/gce_keys/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no" \
+  --ssh-options "--project=cadvisor-e2e" \
```
Comment (krzyzacy) [inline build/jenkins_e2e.sh]: add `--zone` as well

**Comment by dashpole** [comment]
`Required 'compute.instances.get' permission for 'projects/cadvisor-e2e/zones/us-central1-f/instances/e2e-cadvisor-ubuntu-trusty'`

File: build/jenkins_e2e.sh [hunk 4]
```diff
@@ -61,7 +59,15 @@ golden_nodes=(
 
 max_retries=8
 
+if [[ -a "${CADVISOR_SERVICE_ACCOUNT:-}" ]] ; then
```
Comment (krzyzacy) [inline build/jenkins_e2e.sh]: lol the env binded was `CADVISOR_APPLICATION_CREDENTIALS`

File: build/jenkins_e2e.sh [hunk 5]
```diff
@@ -61,7 +55,25 @@ golden_nodes=(
...
+  cp ${JENKINS_GCE_SSH_PUBLIC_KEY_FILE} ~/.ssh/google_compute_engine.pub
+fi
+
 ./runner --logtostderr --test-retry-count=$max_retries \
   --test-retry-whitelist=integration/runner/retrywhitelist.txt \
-  --ssh-options "-i /var/lib/jenkins/gce_keys/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no" \
+  --ssh-options "--project=cadvisor-e2e --strict-host-key-checking=no --zone=us-central1-f" \
   ${golden_nodes[*]}
+
+if [[ -a "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]] ; then
```
Comment (krzyzacy) [inline build/jenkins_e2e.sh]: also you want to switch back the service account regardless of test results

Comment (dashpole) [reply build/jenkins_e2e.sh]: Any tips on how to do that?

Comment (krzyzacy) [reply build/jenkins_e2e.sh]: hummm, you can store exit value from runner and exit with that in the end, and don't `set -e`?

**Comment by BenTheElder** [comment]
You could use a trap for exit instead.

**Comment by BenTheElder** [comment]
But really why doesn't bootstrap just always switch back after the scenario
Sen?

**Comment by krzyzacy** [comment]
@BenTheElder hummm, yeah sure this probably should be handled by bootstrap

**Comment by krzyzacy** [comment]
feel free to drop the switch account back part, we can make a bootstrap change

**Comment by dashpole** [comment]
/retest
gce enforcer save us

**Comment by BenTheElder** [comment]
Still SSH connection timeout and then log upload fails because we haven't
patched bootstrap yet

**Comment by krzyzacy** [comment]
GREEN!

**Comment by BenTheElder** [comment]
> GREEN!

:tada: about time. nice work all :-)
Only 44 @zSys-bot comments later...

**Comment by dashpole** [comment]
(just so you know, its only green because I removed the integration tests from prow_e2e.sh... it only builds cadvisor, and builds the integration tests)

**Comment by dashpole** [comment]
Whoops, I had added kubekins instead of pr-kubekins to the project

**Pull request merged**